File: src/main/java/eu/happybe/openapi/ranks/Rank.java
package eu.happybe.openapi.ranks;

import lombok.Getter;

public class Rank {

    @Getter
    private final String rankName;
    @Getter
    private final String displayFormat;
    @Getter
    private final String[] permissions;
    @Getter
    private final boolean isVisible;

    public Rank(String rankName, String displayFormat, String[] permissions) {
        this(rankName, displayFormat, permissions, true);
    }

    public Rank(String rankName, String displayFormat, String[] permissions, boolean isVisible) {
        this.rankName = rankName;
        this.displayFormat = displayFormat;
        this.permissions = permissions;
        this.isVisible = isVisible;
    }

    @Deprecated
    public String getFormatForChat() {
        return this.getDisplayFormat();
    }

    @Deprecated
    public String getFormatForDisplay() {
        return this.getDisplayFormat();
    }

    @Deprecated
    public String getFormatForNameTag() {
        return this.getDisplayFormat();
    }
}

File: src/main/java/eu/happybe/openapi/ranks/Rank.kt
package eu.happybe.openapi.ranks

class Rank @JvmOverloads constructor(@field:Getter private val rankName: String, @field:Getter private val formatting: String, @field:Getter private val permissions: Array<String>, @field:Getter private val isVisible: Boolean = true) {
    val formatForChat: String
        get() = if (isVisible) this.getFormatting() + this.getRankName().toUpperCase() + "§r " else ""
    val formatForDisplay: String
        get() = this.getRankName()
    val formatForNameTag: String
        get() = if (isVisible) this.getFormatting() + this.getRankName().toUpperCase() + "§r " else ""
}